turbopack/crates/turbopack-ecmascript/src/references/esm/binding.rs

```diff
@@ -149,46 +149,37 @@ impl EsmBinding {
                     ));
                     break;
                 }
-                Some(swc_core::ecma::visit::AstParentKind::BindingIdent(
-                    swc_core::ecma::visit::fields::BindingIdentField::Id,
-                )) => {
+                // We need to handle LHS because of code like
+                // (function (RouteKind1){})(RouteKind || RouteKind = {})
+                Some(swc_core::ecma::visit::AstParentKind::SimpleAssignTarget(_)) => {
                     ast_path.pop();
 
-                    // We need to handle LHS because of code like
-                    // (function (RouteKind1){})(RouteKind || RouteKind = {})
-                    if let Some(swc_core::ecma::visit::AstParentKind::SimpleAssignTarget(
-                        swc_core::ecma::visit::fields::SimpleAssignTargetField::Ident,
-                    )) = ast_path.last()
-                    {
```

Comment:
We can't just match on `SimpleAssignTargetField::Ident`. We need to rewrite

```
import * as esm from 'esm';
esm.foo = bar;
```

This assignment isn't valid, but it's supposed to cause a runtime error because the field has no setter, so it should still codegen.

There's an execution test that covers the runtime exception, which was only working before because the analysis bailed out on `SimpleAssignTarget`.

The end result is similar (a runtime error), but now the generated code makes a bit more sense, IMO.